tests/test_configuration.yml

```diff
@@ -0,0 +1,33 @@
+---
+
+- name: copy load-balancer configuration fragment to the server
+  copy:
+    content: "{{ config_fragment }}"
+    dest: /tmp/config-fragment
+- name: copy load-balancer backend map fragment to the server
+  copy:
+    content: "{{ backend_map_fragment }}"
+    dest: /tmp/backend-map-fragment
+- name: apply the new configuration
+  command: haproxy-config apply test /tmp/config-fragment /tmp/backend-map-fragment
+- name: wait for the certificate to be available
```

Comment:
@pomegranited The version from `master` doesn't clean up old certificates properly, and the files that get left behind interfere with the next integration test run and cause the additional `-0001` in the file name.  (Note that this wouldn't be a problem in production if it happened; it's only a problem since we are waiting for a file with the exact name `<domain>.pem` in the integration test.  I would like to have a more flexible condition in the integration test, but Ansible's `wait_for` is rather limited.)

This problem has been fixed on this branch, but since you ran the playbook on master first, you still experienced it for now.  To make the tests more robust, I added some clean-up code to the playbook that removes any left-over files first.  This shouldn't be necessary in the future, but it doesn't hurt.

The second error indicates that the backend server, i.e. `pydoc`, has died in the meantime.  I've got no idea why this happened.  It is configured to run for ten minutes, which should be enough.  I simply upped this time to one hour;  pydoc is killed when Ansible closes the SSH connection anyway.